tests/ir.at

```diff
@@ -0,0 +1,22 @@
+AT_BANNER([intermediate representation unit tests])
+
+m4_define([TEST_IR_JSON_DUMP],
+[AT_SETUP([$1])
+ AT_CHECK([p4ctest $2], [0], [stdout], [])
+ AT_CLEANUP]
+)
+TEST_IR_JSON_DUMP([ir - dump json], [test-json])
+
+m4_define([TEST_IR_OPEQ],
+[AT_SETUP([$1])
+ AT_CHECK([p4ctest $2], [0], [stdout], [])
+ AT_CLEANUP]
+)
+TEST_IR_OPEQ([ir - opeq ], [test-opeq])
+
+m4_define([TEST_IR_TRANSFORM],
+[AT_SETUP([$1])
+ AT_CHECK([p4ctest $2], [0], [stdout], [])
+ AT_CLEANUP]
+)
+TEST_IR_OPEQ([ir - transform ], [test-transform])
```

Comment:
This seems like a lot of boilerplate -- why define a new TEST_XXX macro for each test that you then invoke once, particularly since they're all the same.  Why not just a single macro invoked mulitple times (once for each test).